src/main/java/com/google/firebase/messaging/Aps.java

```diff
@@ -137,6 +146,41 @@ public Builder setContentAvailable(boolean contentAvailable) {
       return this;
     }
 
+    /**
+     * Specifies whether to set the {@code mutable-content} property on the message, so the
+     * clients can modify the notification via app extensions.
+     *
+     * @param mutableContent True to make the content mutable via app extensions.
+     * @return This builder.
+     */
+    public Builder setMutableContent(boolean mutableContent) {
+      this.mutableContent = mutableContent;
+      return this;
+    }
+
+    /**
+     * Puts a custom key-value pair to the aps dictionary.
+     *
+     * @param key A non-null key.
+     * @param value A non-null, json-serializable value.
+     * @return This builder.
+     */
+    public Builder putCustomData(@NonNull String key, @NonNull Object value) {
+      this.customData.put(key, value);
+      return this;
+    }
+
+    /**
+     * Puts all the key-value pairs in the specified map to the aps dictionary.
+     *
+     * @param fields A non-null map. Map must not contain null keys or values.
+     * @return This builder.
+     */
+    public Builder putAllCustomData(@NonNull Map<String, Object> fields) {
```

Comment:
Java 8 allows you to do `@NonNull Map<@NonNull String, @NonNull Object>`

Reply:
We are still on Java 7. Only GAE support has been bumped to Java 8.